pkg/cache/internal/cache_reader.go

```diff
@@ -125,7 +125,9 @@ func (c *CacheReader) List(_ context.Context, out runtime.Object, opts ...client
 		if !isObj {
 			return fmt.Errorf("cache contained %T, which is not an Object", obj)
 		}
-		runtimeObjs = append(runtimeObjs, obj)
+		outObj := obj.DeepCopyObject()
```

Comment:
I don't believe that we need to deep copy here because these objects are all created from within this function? 

Thoughts?

Reply:
I added the `DeepCopy` because it stopped the race detector going off when I was running tests so there must be something funky going on here but I'm not sure what 🤷‍♂️

Reply:
👍 wonder if it is what the indexer gives us for each list is just the cached object? I will look it up to see eventually :) thanks for the explanation!

Reply:
I think we should've been deepcopying anyway, since that's what we do elsewhere anyway.